### DISCUSSION TITLE
File Protocol Error when using export

### DISCUSSION THREAD

**Discussion by kdev** [ivanhofer/typesafe-i18n#600] [Q&A] [answered]
So I am trying to use `export` but encounter the following error:
```
[typesafe-i18n] exporting translations for locale 'de' ...
[typesafe-i18n] exporting translations for locale 'en' ...
[typesafe-i18n] ERROR: import failed for E:\<redacted>\node_modules\typesafe-i18n\temp-output\en\en\index.js Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'e:'
[typesafe-i18n] ERROR: import failed for E:\<redacted>\node_modules\typesafe-i18n\temp-output\de\de\index.js Error [ERR_UNSUPPORTED_ESM_URL_SCHEME]: Only URLs with a scheme in: file, data are supported by the default ESM loader. On Windows, absolute paths must be valid file:// URLs. Received protocol 'e:'
[typesafe-i18n] ERROR: could not read default export from base locale file 'en'
[typesafe-i18n] ERROR: could not find locale file 'en'
E:\<redacted>\node_modules\typesafe-i18n\exporter\index.cjs:7881
  throw new TypesafeI18nParseError();
        ^
TypesafeI18nParseError
    at throwError (E:\<redacted>\node_modules\typesafe-i18n\exporter\index.cjs:7881:9)
    at readTranslation (E:\<redacted>\node_modules\typesafe-i18n\exporter\index.cjs:8379:13)
    at async Promise.all (index 1)
    at async exportTranslationsForLocale (E:\<redacted>\src\i18n\export.ts:14:21)
```
This looks like an error on typesafes side with ESM loader confusing `e:` with the file protocol

I am using node 18 and typescript v4.9.5 on a Win10 machine.

**Comment by ivanhofer** [comment, marked as answer]
~`E:\<redacted>\node_modules\typesafe-i18n\temp-output\en\en\index.js`
There are 2 `en` slugs at the end of the path.. maybe that's the issue.~ _edit: didn't read the full error message before ^^_

Can you create a small repo and share it with me?

**Reply by kdev** [reply]
I am using the example from the `exports` package and `tsx` to execute the file (I've also tried `ts-node` with the same result

**Reply by ivanhofer** [reply]
There is a CI process that checks every day if the example from `exports` works: https://github.com/ivanhofer/typesafe-i18n/blob/main/.github/workflows/example-typesafe-i18n-exporter.yml

And it did run successfully a few hours ago

**Reply by kdev** [reply]
Here ist a reproducable example: https://github.com/kdev/typesafe-i18n-unsupported-url-scheme

**Reply by ivanhofer** [reply]
#worksOnMyMachine

I tried it with node `14`, `16` and `18` on Ubuntu 22.04. So the issue only occurs on windows.
Probably this line is the problem: https://github.com/ivanhofer/typesafe-i18n/blob/7ab228f2059e0af58e367b380bbe81167278cc91/packages/generator/src/utils/file.utils.mts#L119
Can you experiment a bit and see what works? It should be line `7956` in `./node_modules/typesafe-i18n/exporter/index.cjs`

**Reply by kdev** [reply]
There seems to be a problem with the condition `isWindows && isEsm` returning false.
I'm still investigating tho

**Reply by kdev** [reply]
So it works when just using `pathToFileURL(file).href` but I cannot modify the `isEsm` condition to return true as I do not know how Esm and win32 are connected

**Reply by ivanhofer** [reply]
I probably habe added the `isEsm` check at the time where it was experimental in node. I would need to check that again but I think we can get rid of this extra check.

**Reply by ivanhofer** [reply]
I got rid of the `isEsm` check and have released version `5.24.2`. Thanks for your help!

**Reply by kdev** [reply]
Thanks to you for resolving this issue so quickly!
